### DISCUSSION TITLE
Next.js middleware not running twice for same route

### DISCUSSION THREAD

**Discussion by andregosling** [vercel/community#1490] [Help]
### Summary

Hi guys! I'm having one issue here with my Next.js middleware, and here goes my situation: 

I have a route to login my customer (/customer/auth), and one page for my customers be able to see they orders (/customer/orders). The middleware redirects the user trying to access the order page to the auth page if the user isn't authenticated, and redirects from the auth page to the order page if the user is already authenticated. 

This works fine, but there is one specific use-case that brokes it. 

Basically, if the user sign in, go to orders page, and try to get back to auth page, without any refresh on the window, the middleware is not called again on auth page, and user can access. This behavior only happens on production, never in localhost, and is fixed after the user refresh's the page.

Any ideias what it can be?
